### DISCUSSION TITLE
unable to push my local repo to Github repo

### DISCUSSION THREAD

**Discussion by ghost** [community/community#52796] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

is now more than 2 months that everytime that i try to push my repository online i will receive this: << fatal: Authentication failed for 'https://github.com/pdjamen88/terraform-dev.git/' >> but when i generated a key token i will be able to push after entering the generated key token as password. and everything that i want to push i will need to do the same process. so please i would like to know how can i fix it in such way that i will not more have stress. thank you

**Comment by airtower-luna** [comment]
> but when i generated a key token i will be able to push after entering the generated key token as password. and everything that i want to push i will need to do the same process.

I think what you mean is that creating and pasting the token is too much trouble? There are two parts to the answer:

* You don't need to create a new token every time. You can choose how long it should be valid (e.g. 3 months) and store it in a password manager, and just copy it from there when you need it.
* You can tell Git to cache credentials (username and token) you have entered for a little while, see [Git Tools - Credential Storage](https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage). There's also the option to have Git store the credentials permanently, but that would write the token to disk in plain text, which is a security risk.
